mapie/classification.py

```diff
@@ -889,83 +887,86 @@ def predict(
+                            y_pred_included > _alpha * (n - 1)
                             for _alpha in alpha_
                         ], axis=2
                     )
 
             elif self.method in ["cumulated_score", "naive"]:
                 if (cv == "prefit") or (agg_scores in ["mean"]):
-                    # sort labels by decreasing probability
-                    index_sorted = np.fliplr(np.argsort(y_pred_proba, axis=1))
-                    # sort probabilities by decreasing order
-                    y_pred_proba_sorted = np.take_along_axis(
-                        y_pred_proba, index_sorted, axis=1
-                    )
-                    # get sorted cumulated score
-                    y_pred_proba_sorted_cumsum = np.cumsum(
-                        y_pred_proba_sorted, axis=1
+                    thresholds = self.quantiles_
+                else:
+                    thresholds = self.conformity_scores_.ravel()
+                # sort labels by decreasing probability
+                index_sorted = np.flip(
+                    np.argsort(y_pred_proba, axis=1), axis=1
+                )
+                # sort probabilities by decreasing order
+                y_pred_proba_sorted = np.take_along_axis(
+                    y_pred_proba, index_sorted, axis=1
+                )
+                # get sorted cumulated score
+                y_pred_proba_sorted_cumsum = np.cumsum(
+                    y_pred_proba_sorted, axis=1
+                )
+                # get cumulated score at their original position
+                y_pred_proba_cumsum = np.take_along_axis(
+                    y_pred_proba_sorted_cumsum,
+                    np.argsort(index_sorted, axis=1),
+                    axis=1
+                )
+                # get index of the last included label
+                y_pred_index_last = self._get_last_index_included(
+                    y_pred_proba_cumsum,
+                    thresholds,
+                    include_last_label
+                )
+                # get the probability of the last included label
+                y_pred_proba_last = np.take_along_axis(
+                    y_pred_proba,
+                    y_pred_index_last,
+                    axis=1
+                )
+                # get the prediction set by taking all probabilities
+                # above the last one
+                if (cv == "prefit") or (agg_scores in ["mean"]):
+                    y_pred_included = (
+                        (y_pred_proba >= y_pred_proba_last - EPSILON)
                     )
-                    # get cumulated score at their original position
-                    y_pred_proba_cumsum = np.take_along_axis(
-                        y_pred_proba_sorted_cumsum,
-                        np.argsort(index_sorted),
-                        axis=1
+                else:
+                    y_pred_included = (
+                        ~(y_pred_proba >= y_pred_proba_last - EPSILON)
```

Comment:
Why this negation ?

Reply:
In order to be consistent with the formula 3 lines above but it does not make sense to have a double negation.